app/routes.tsx

```diff
@@ -101,9 +102,21 @@ export const routerConfig = [
     path: routes.STACKING,
     component: DirectStacking,
   },
+  {
+    path: routes.UNLOCK,
+    component: Unlock,
+  },
 ];
 
-const getAppStartingRoute = (address?: string) => (!!address ? routes.HOME : routes.TERMS);
+const getAppStartingRoute = (publicKey: Buffer | null, initialized: boolean) => {
+  if (!initialized) {
+    return routes.TERMS;
+  }
+  if (!publicKey) {
+    return routes.UNLOCK;
+  }
```

Comment:
afaict this is the source of the bug. This needs to come first, as if I already have a wallet, it isn't `initialised`, so it takes me to terms.